src/app/proposals/shared/utils.ts

```diff
@@ -153,3 +155,57 @@ export const isChecksumValid = (value: string, chainId?: string) => {
     checksumAddress(value as Address, chainId ? Number(chainId) : undefined) === value
   )
 }
+
+type ProposalSource = 'DAO' | 'TALLY' | 'UNKNOWN'
+interface ParsedDescription {
+  name: string
+  description: string
+  source: ProposalSource
+}
+
+export const parseProposalDescription = (description: string): ParsedDescription => {
+  // Constants
+  const MAX_NAME_LENGTH = 100
+  const TALLY_DOUBLE_SPACE = '  ' // Tally uses double spaces to separate name and description
+
+  // Default result
+  let result: ParsedDescription = {
+    name: '',
+    description: description,
+    source: 'UNKNOWN',
+  }
+
+  // Check if it's our DAO format (contains semicolon)
```

Comment:
The comment has to be changed to:
"If the proposal description contains semicolon, we will automatically assume it's ours (for now)"

Because I might create a proposal from Tally that has a semicolon and it'll be picked it by this condition.